src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -583,39 +550,15 @@ Namespace Microsoft.CodeAnalysis.Operations
                 End If
             End If
 
-            If boundConversion.ExplicitCastInCode AndAlso conversion.IsIdentity AndAlso
-               boundOperand.Kind = BoundKind.Conversion Then
-                Dim nestedConversion = DirectCast(boundOperand, BoundConversion)
-                Dim nestedConversionOperandAndMethod As (Operand As BoundExpression, Method As MethodSymbol) = GetConversionInfo(nestedConversion)
-                Dim nestedOperand As BoundExpression = nestedConversionOperandAndMethod.Operand
-
-                If nestedConversion.Syntax Is nestedOperand.Syntax AndAlso nestedConversion.ExplicitCastInCode AndAlso
-                   nestedOperand.Kind = BoundKind.ConvertedTupleLiteral AndAlso
-                   nestedConversion.Type <> nestedOperand.Type Then
-                    ' Let's erase the nested conversion, this is an artificial conversion added on top of BoundConvertedTupleLiteral
-                    ' in Binder.ReclassifyTupleLiteral.
-                    ' We need to use conversion information from the nested conversion because that is where the real conversion 
-                    ' information is stored.
-                    conversionOperandAndMethod = nestedConversionOperandAndMethod
-                    conversion = CreateConversion(nestedConversion)
-                    boundOperand = nestedOperand
-                End If
-            End If
-
-            Dim conversionInformation = CreateConversionOperand(boundOperand, conversion.Kind, boundConversion.Syntax, boundConversion.Type)
-            Dim methodSymbol As MethodSymbol = conversionOperandAndMethod.Method
-            Dim operand As Lazy(Of IOperation) = conversionInformation.Operation
-            Dim isDelegateCreation As Boolean = conversionInformation.IsDelegateCreation
+            Dim conversionInfo = GetConversionInfo(boundConversionOrCast)
+            Dim conversion As Conversion = conversionInfo.ConversionStruct
```

Comment:
>Dim conversion As Conversion = conversionInfo.ConversionStruct [](start = 12, length = 62)

Consider getting rid of the local, or moving its declaration into the ```Else``` block below.